core/src/storage/lazy/lazy_chunk.rs

```diff
@@ -0,0 +1,384 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::super::{
+    KeyPtr,
+    PullForward,
+    PushForward,
+    StorageSize,
+};
+use core::cell::UnsafeCell;
+use ink_primitives::Key;
+
+use ink_prelude::{
+    boxed::Box,
+    collections::BTreeMap,
+};
+
+/// The index type used in the lazy storage chunk.
+pub type Index = u32;
+
+/// A lazy storage chunk that spans over a whole chunk of storage cells.
+///
+/// # Note
+///
+/// This is mainly used as low-level storage primitives by other high-level
+/// storage primitives in order to manage the contract storage for a whole
+/// chunk of storage cells.
+///
+/// A chunk of storage cells is a contiguous range of 2^32 storage cells.
+#[derive(Debug, Default)]
```

Comment:
```suggestion
#[derive(Default)]
```

Reply:
There are a number of other `Debug` derives in the PR, I haven't annotated them in the review, but you might want to do a full text search.

Reply:
I have to think about the usefulness of those `Debug` derives in the context of our new off-chain environment. Maybe it actually makes sense to have it since we can assume dead code elimination for `no_std` (e.g. Wasm) builds.